drv/ignition-server/src/main.rs

```diff
@@ -64,6 +64,25 @@ fn main() -> ! {
 
     // Determine the number of Ignition controllers available.
     server.port_count = server.controller.port_count().unwrap_lite();
+
+    // Starting with rev C the Ignition Controller has a 36th link to the Target
+    // on its own board, allowing the control plane to query for full rack
+    // presence via either Sidecar. The mainboard controller of rev B boards
+    // does implement this 36th link but it is not actually connected to the
+    // Target. In order to make this explicit the port count is adjusted down to
+    // 35 so anything querying a Sidecar can distinguish between a rev B and a
+    // rev C with a faulty local link.
+    //
+    // Methods provided by this server use the count as an upper bound when
+    // iterating over the ports, causing port 35 to be ignored on rev B
+    // systems.
+    #[cfg(target_board = "sidecar-b")]
+    {
+        if server.port_count == 36 {
+            server.port_count = 35;
+        }
+    }
```

Comment:
```suggestion
    if cfg!(target_board = "sidecar-b") && server.port_count == 36 {
        server.port_count = 35;
    }
```

Also, why does `port_count` return 36 on the rev B bitstream, rather than 35?

Reply:
Thanks, this is much cleaner.

I also added to the comment to try and explain better why this is done here. The tl;dr is that this uses a set of templates to generate the RTL which involve some shenanigans to make work if there are 36 sets of device pins but only 35 channels. Performing a small three line workaround here seems like a reasonable trade-off.

Reply:
I moved this into `drv/sidecar-mainboard-controller` since its cleaning up a hardware detail. This leaves the `ignition-server` task with less implementation details.